### PR TITLE
initial effort at enabling Windows with ARM template - added config o…

### PR DESCRIPTION
**Pull request by rysweet**
…ptions for admin user and pass; added NSG rules for WinRM and RDP; - tested with     azure.vm_image_urn = MicrosoftVisualStudio:VisualStudio:VS-2015-ENT-VSU2-AzureSDK-29-W10T-N-x64:latest

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@rysweet__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: lib/vagrant-azure/action/run_instance.rb [hunk 1]
```diff
@@ -85,12 +95,15 @@ def call(env)
             paths_to_pub = private_key_paths.map{ |k| File.expand_path( k + '.pub') }.select{ |p| File.exists?(p) }
             raise I18n.t('vagrant_azure.public_key_path_private_key', private_key_paths.join(', ')) if paths_to_pub.empty?
             deployment_params.merge!(sshKeyData: File.read(paths_to_pub.first))
+          else
+            windows_params = {
+              adminUsername:  adminUsername,
+              adminPassword:  adminPassword,    
+            }
+            deployment_params.merge!(windows_params)
+            @logger.info(pp(deployment_params))
```
Comment (matt-richardson) [inline lib/vagrant-azure/action/run_instance.rb:104]: This will dump the password in plain text

**Comment by matt-richardson** [comment]
Hey, @rysweet, can you sign the CLA? I'm happy to take what you've started and finish it off, but AFAIK, you need to sign the CLA for me to be able to use it.

**Comment by devigned** [comment]
@matt-richardson here's the last ping from @rysweet.

> I have been spending time on other things but info want to push to get this vagrant azure plugin work complete. 
>
> I got the winrm communicator to attempt to connect, but for the life of me it is refusing to acknowledge any options to set the destination port to anything other than 5985. I can get it to try SSL, but it won't accept any port configuration.  Have you experienced that? 

Let me know if you have any questions. Please ping if you have any questions.

**Comment by rysweet** [comment]
I thought I did sign the CLA?

If that didn’t take point me to the procedure and no problem.  I had to walk away from this work as other priorities emerged.

-Ryan

**Comment by matt-richardson** [comment]
Hmmm. Not sure. I'm just going by the fact that the label 'cla-required is sill there'. @devigned - any ideas?

I've not signed it yet myself, so not sure how the process works. All I know is the URL is https://cla.azure.com/ :)

I've got a local copy working over winrm (well, except for there's some weird timeout that I'm still looking into). I'll get a WIP PR up in the next day or so.

**Comment by devigned** [comment]
@matt-richardson if you open up your own PR, you will be prompted to sign the CLA. I or someone else will accept your signature and then the PR will be marked green. Should be easy peasy.

**Pull request closed without merging**